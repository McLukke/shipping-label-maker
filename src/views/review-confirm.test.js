import { render, screen, act, waitFor, within } from "@testing-library/react";
import ShallowRenderer from "react-test-renderer/shallow";
import renderer from "react-test-renderer";
import user from "@testing-library/user-event";

import ReviewConfirm from "./review-confirm";

describe("Review & Confirm", () => {
  let props;

  beforeEach(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });

    props = {
      shippingLabel: {
        from: {
          address: "111",
          city: "Holtsville",
          name: "111",
          state: "New York",
          zipCode: 501,
        },
        to: {
          address: "222",
          city: "Aguada",
          name: "222",
          state: "Puerto Rico",
          zipCode: 602,
        },
        ReviewConfirm: 2,
        weight: "333",
      },
      resetAppState: jest.fn(),
      setCurrentStep: jest.fn(),
      currentStep: 4,
    };

    render(<ReviewConfirm {...props} />);
  });

  it("renders without crashing", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ReviewConfirm {...props} />);
    const view = renderer.getRenderOutput();

    expect(view).toBeTruthy();
  });

  it("snapshots to match", () => {
    const component = renderer.create(<ReviewConfirm {...props} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
