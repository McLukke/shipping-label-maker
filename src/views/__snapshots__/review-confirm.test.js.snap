// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review & Confirm snapshots to match 1`] = `
<div
  className="ant-row ant-row-center ant-row-middle"
  style={Object {}}
>
  <div
    className="ant-col ant-col-24"
    style={Object {}}
  >
    <h3
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Review and Confirm:
    </h3>
  </div>
  <div
    className="ant-col ant-col-xs-24 ant-col-sm-20 ant-col-md-18 ant-col-lg-16 ant-col-xl-12"
    style={Object {}}
  >
    <pre
      className="code"
    >
      <code>
        {
    "from": {
        "address": "111",
        "city": "Holtsville",
        "name": "111",
        "state": "New York",
        "zipCode": 501
    },
    "to": {
        "address": "222",
        "city": "Aguada",
        "name": "222",
        "state": "Puerto Rico",
        "zipCode": 602
    },
    "ReviewConfirm": 2,
    "weight": "333"
}
      </code>
    </pre>
    <h5
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Your total shipping cost is: 
      $199.80
    </h5>
    <div
      className="ant-space ant-space-horizontal ant-space-align-center"
      style={Object {}}
    >
      <div
        className="ant-space-item"
        style={
          Object {
            "marginRight": 8,
          }
        }
      >
        <button
          className="ant-btn ant-btn-default"
          onClick={[Function]}
          type="button"
        >
          <span>
            Back
          </span>
        </button>
      </div>
      <div
        className="ant-space-item"
        style={Object {}}
      >
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="button"
        >
          <span>
            Confirm
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
